src/test/java/nc/students/ayaz/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,75 @@
+package nc.students.ayaz.controllers;
+
+import nc.students.ayaz.model.User;
+import nc.students.ayaz.model.Video;
+import nc.students.ayaz.model.exceptions.NoSuchUserException;
+import nc.students.ayaz.repositories.UserRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
+
+    @MockBean
+    private UserRepository repository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Test
+    public void shouldAddUserToRepositoryWhenPostRequestSent() {
+        ResponseEntity<String> response = restTemplate.postForEntity("/users/Ayaz", "", String.class);
+
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    }
+
+    @Test
+    public void shouldReturnUserVideosWhenGetRequestSentWithUsersNickname() throws Exception {
+        User user = new User("Ayaz");
+        Video firstVideo = new Video("FunnyCats");
+        Video secondVideo = new Video("SadDogs");
```

Comment:
&#x1F34E; **Issue**
We don't need a second video here. It doesn't change anything and only makes this test harder to read. (You are asserting that list contains an element, one element will do just fine for such a check. You might also consider asserting on the size of your list.)